### ISSUE TITLE
[Question FBGEMM_GPU] Adam optmizer not optimized

### ISSUE THREAD

**Issue by JacoCheung** [pytorch/FBGEMM#2824]
Hi team, I'm using Adam optimizer for my model. But there is a warning regarding performance. (Can It be resolved? Or do you have any quantitative number for the perf degradation?)

```
[FBGEMM_GPU] NOTE: The training optimizer 'adam' is marked as
        EXPERIMENTAL and thus not optimized, in order to reduce code compilation
        times and build sizes!
```

I also noted that there was a [discussion](https://github.com/pytorch/FBGEMM/discussions/1727) about the optimizer.
It seemed that adam was not considered for optimizations.  I'd like to know what's the plan for Adam for today.  Thanks!

**Comment by JacoCheung** [comment]
Another issue is when I specify the ouput datatype as bf16, it hit me with not implemented error.

**Comment by JacoCheung** [comment]
Hi @sryap , thanks for your reply. I'll try this flag out.

Re BF16, it seems that the error is rasied by forward kernel (regardless of optimzer). 
I was using v6.0.0.  I checked [changelog](https://github.com/pytorch/FBGEMM/blob/v0.6.0/fbgemm_gpu/include/fbgemm_gpu/dispatch_macros.h#L72-L99) just now and found out **it's supported since v7.0.0**.

**Comment by JacoCheung** [comment]
Regarding the fp16 output dtype, fbgemm does not have a scaler for backward/update. Is this intended?

**Comment by JacoCheung** [comment]
The scaler used in mxied precision training.

**Comment by JacoCheung** [comment]
Sorry for my confusion. Let me clarify a little bit.

The scalar I refer to is a generic concept in [mixed-precision training ](https://pytorch.org/docs/stable/notes/amp_examples.html) esp in fp16 training. In fp16 training schema, the loss is usually scaled, and so the dgrad  is scaled in the bwd. There should be a unscaling process for wgrad. 

However, fbgemm_gpu fuses update with bwd / dgrad (TBE does not have explict wgrad ). So I expect the forward() function of TBE operator to accept a scaling factor, and do the dgrad/wgrad unscaling at backward stage.
